src/code_builder.rs

```diff
@@ -1288,74 +1254,64 @@ fn f64_store<C: Config>(
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
-    let offset = u.arbitrary()?;
-    let align = *u.choose(&[0, 1, 2])?;
     builder.pop_operands(&[ValType::I32, ValType::F64]);
-    Ok(Instruction::F64Store(MemArg { offset, align }))
+    Ok(Instruction::F64Store(mem_arg(u, builder, &[0, 1, 2])?))
 }
 
 fn i32_store_8<C: Config>(
     u: &mut Unstructured,
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
-    let offset = u.arbitrary()?;
-    let align = 0;
     builder.pop_operands(&[ValType::I32, ValType::I32]);
-    Ok(Instruction::I32Store8(MemArg { offset, align }))
+    Ok(Instruction::I32Store8(mem_arg(u, builder, &[0])?))
 }
 
 fn i32_store_16<C: Config>(
     u: &mut Unstructured,
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
-    let offset = u.arbitrary()?;
-    let align = *u.choose(&[0, 1])?;
     builder.pop_operands(&[ValType::I32, ValType::I32]);
-    Ok(Instruction::I32Store16(MemArg { offset, align }))
+    Ok(Instruction::I32Store16(mem_arg(u, builder, &[0, 1])?))
 }
 
 fn i64_store_8<C: Config>(
     u: &mut Unstructured,
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
-    let offset = u.arbitrary()?;
-    let align = 0;
     builder.pop_operands(&[ValType::I32, ValType::I64]);
-    Ok(Instruction::I64Store8(MemArg { offset, align }))
+    Ok(Instruction::I64Store8(mem_arg(u, builder, &[0])?))
 }
 
 fn i64_store_16<C: Config>(
     u: &mut Unstructured,
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
-    let offset = u.arbitrary()?;
-    let align = *u.choose(&[0, 1])?;
     builder.pop_operands(&[ValType::I32, ValType::I64]);
-    Ok(Instruction::I64Store16(MemArg { offset, align }))
+    Ok(Instruction::I64Store16(mem_arg(u, builder, &[0, 1])?))
 }
 
 fn i64_store_32<C: Config>(
     u: &mut Unstructured,
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
-    let offset = u.arbitrary()?;
-    let align = *u.choose(&[0, 1, 2])?;
     builder.pop_operands(&[ValType::I32, ValType::I64]);
-    Ok(Instruction::I64Store32(MemArg { offset, align }))
+    Ok(Instruction::I64Store32(mem_arg(u, builder, &[0, 1, 2])?))
 }
 
 fn memory_size<C: Config>(
-    _: &mut Unstructured,
+    u: &mut Unstructured,
     _: &ConfiguredModule<C>,
     builder: &mut CodeBuilder<C>,
 ) -> Result<Instruction> {
     builder.push_operands(&[ValType::I32]);
-    Ok(Instruction::MemorySize)
+    Ok(Instruction::MemorySize(
+        u.int_in_range(0..=builder.allocs.num_memories - 1)?,
```

Comment:
As an optimization for using up less of our entropy when possible, if there is only one memory then let's avoid calling `u.int_in_range`.